### PR TITLE
Fix null namespace for FunctionReference__mdt lookup

### PR DESCRIPTION
**Pull request by cwallsfdc**

### CODE REVIEW

**Comment by jjsfdc** [comment]
two issues:

first, unit tests fail, summary: 
```
% sf apex run test

NAME                 VALUE                                                   
───────────────────  ────────────────────────────────────────────────────────
Outcome              Failed                                                  
Tests Ran            53                                                      

...

TEST NAME                                                                                         OUTCOME  MESSAGE                                                                                    RUNTIME (MS)
────────────────────────────────────────────────────────────────────────────────────────────────  ───────  ─────────────────────────────────────────────────────────────────────────────────────────  ────────────

FunctionInvokeTests.testInvoke_async_unhandledExceptionOnInvoke                                   Fail     System.QueryException: Insufficient permissions: secure query included inaccessible field  
                                                                                                           Class.AsyncFunctionInvocationQueueable.execute: line 87, column 1                          
                                                                                                           Class.FunctionInvokeTests.testInvoke_async_unhandledExceptionOnInvoke: line 319, column 1
```

second, my invocation fails for a null namespace:
```
15:02:27.263 (277476588)|USER_DEBUG|[256]|DEBUG|SELECT APIVersion__c, ConsumerKey__c, Certificate__c, Endpoint__c, PermissionSetOrGroup__c FROM FunctionReference__mdt WHERE DeveloperName = 'functions_project_somefn' AND NamespacePrefix = 'null'
15:02:27.263 (282229260)|USER_DEBUG|[262]|DEBUG|Unable to retrieve FunctionReference__mdt for functions_project.somefn: List has no rows for assignment to SObject
```

Notice that the first debug log statement has a SOQL query that ends in ```NamespacePrefix = 'null'```

That query will work correctly for me if there are no quotes around the ```null```

**Review by jjsfdc — CHANGES_REQUESTED** [review]
See comments on pull request, need unit tests to pass and null fixed in SOQL query

**Comment by cwallsfdc** [comment]
For the `testInvoke_async_unhandledExceptionOnInvoke` failure, the invoking user needs permission to `AsyncFunctionInvocationRequest__c`.  A future edit could create a user and temp add them (insert `PermissionSetAssignment`) to the "Functions Authentication" PermSet.  For now, including a comment to test and in `AsyncFunctionInvocationQueueable`.

Changed `FunctionReference__mdt` query from `NamespacePrefix == 'null'` to `NamespacePrefix == ''` for non-namespaced orgs.

cc @jjsfdc

**Review by jjsfdc — APPROVED** [review]
this now works correctly with a null namespace and it deploys successfully into production

**Comment by cwallsfdc** [comment]
@jjsfdc, agree about `System.runAs()`.  Can you approve this PR?  Thank you.

**Comment by jjsfdc** [comment]
I have approved the PR, but it looks like you will need someone else to approve as well, for I don't have write access.

File: force-app/main/default/classes/functions/AsyncFunctionInvocationQueueable.cls [hunk 1]
```diff
@@ -74,6 +74,10 @@ public class AsyncFunctionInvocationQueueable implements Queueable, Database.All
 
     /**
      * Finalizer implementation captures failed function invocation jobs on unexpected execptions, eg limit errors.
+     *
+     * Requires that the invoking user is assigned access to AsyncFunctionInvocationRequest__c (likely via
```
Comment (dhagberg-sf) [inline force-app/main/default/classes/functions/AsyncFunctionInvocationQueueable.cls:78]: double-checked and the README includes that information - so requirement already doc'd
> You must assign Edit access to AsyncFunctionInvocationRequest__c for invoking users. You can do so either with the FunctionReference.PermissionSetOrGroup__c‘s permission set or another permission set.

**Review by dhagberg-sf — APPROVED** [review]
✅

**Review by dhagberg-sf — APPROVED** [review]
re- ✅
